internal/services/monitor/monitor_aad_diagnostic_setting_resource.go

```diff
@@ -82,22 +84,18 @@ func resourceMonitorAADDiagnosticSetting() *pluginsdk.Resource {
 				AtLeastOneOf: []string{"eventhub_authorization_rule_id", "log_analytics_workspace_id", "storage_account_id"},
 			},
 
-			"log": {
-				Type:     pluginsdk.TypeSet,
-				Required: true,
+			"enabled_log": {
+				Type:         pluginsdk.TypeSet,
+				Optional:     true,
+				Computed:     !features.FourPointOhBeta(),
+				ExactlyOneOf: []string{"enabled_log", "log"},
```

Comment:
since the `log` property is now behind the `features.FourPointOhBeta()` flag, this would break the schema when the flag is enabled and it is not able to find a property called `log`